### DISCUSSION TITLE
OAuth2 Strategy stucks at /callback route after authorization

### DISCUSSION THREAD

**Discussion by RanjeetEppa18** [nuxt-community/auth-module#1549] [Support Requests]
## Checklist
* [x]  I have tested with the latest Nuxt version and the issue still occurs
* [x]  I have tested with the latest module version and the issue still occurs
* [x]  I have searched the issue tracker and this issue hasn't been reported yet

---

### Steps to reproduce
I would like to use `oauth2` schema in order to fetch `accessToken`,

this is how I set the strategy:
```
linkedIn: {
        scheme: 'oauth2',
        grantType: 'authorization_code',
        responseType: 'code',
        clientId: 'set_me',
        scope: ['r_emailaddress', 'r_liteprofile'],
        endpoints: {
          authorization: 'https://www.linkedin.com/oauth/v2/authorization',
          token: 'https://www.linkedin.com/oauth/v2/accessToken',
          userInfo: 'https://www.linkedin.com/oauth/v2/accessToken'
        }
      }
```
It redirects to `/callback` as expected with `code` as query parameter
e.g `http://localhost:3000/callback?code=code`
And there after nothing happens.

---

**Am I suppose to call `accessToken` linkedIn API inside `callback.vue`?**
I have no Idea what to do inside `callback.vue`

### What is expected?
It should call `accessToken` API implicitly after the response of `authorization` API

**Comment by Tomaszal** [comment]
I am currently struggling with the same issue, were you able to solve it?

**Comment by Intevel** [comment]
**Hello everyone 👋**
With custom OAuth providers you have to send a request to the API yourself in the callback to exchange the code with the access token. With the pre-built providers we have the `addAuthorize` method which do this.

You can set the token you got with:
`this.$auth.setToken("yourstrategy", token);`
